client/signup/controller.js

```diff
@@ -127,9 +127,8 @@ export default {
 	},
 
 	saveInitialContext( context, next ) {
-		if ( ! initialContext ) {
-			initialContext = Object.assign( {}, context );
-		}
+		// Merge the context as some properties are available only on the initial.
+		initialContext = Object.assign( {}, initialContext ?? {}, context );
```

Comment:
This changes the meaning of what `initialContext` is. It's no longer "initial", it gets updated with every navigation.

`initialContext` gets passed to `<SignupComponent>` which it then uses to pre-populate the signup dependencies as expected.

Another way to think about the bug is that the `<SignupComponent>` is only pre-populating signup dependencies based on the "initial" context, when it should really update them based on query parameters that come from all navigation events (not just the initial one).

Reply:
Or I could be thinking about "initial" all wrong. I'm thinking it means "the value of context when the page first loads", but it could also mean "value of context when the signup flow is initialised".

In which case it's ok to write over `initialContext` when the user navigates away from the signup framework and back again.

Reply:
Makes sense to me! It might be better to update initialContext only when the flow changes. Since initialContext represents the initial state of the flow, it should be updated accordingly whenever the flow changes.

Reply:
> only when the flow changes

That's a good one test. So if the user refreshes the page, the `initialContext` would no longer change?

Reply:
Oh I see, it's changed to "either it's a page reload _or_ a flow change". That makes sense.